packages/core/rendering/core-rendering-server-internal/src/rendering_service.tsx

```diff
@@ -218,3 +243,26 @@ const isAuthenticated = (auth: HttpAuth, request: KibanaRequest) => {
   // status is 'unknown' when auth is disabled. we just need to not be `unauthenticated` here.
   return authStatus !== 'unauthenticated';
 };
+
+function isDarkMode(
+  userSettings: {
+    defaults: Readonly<Record<string, Omit<UiSettingsParams, 'schema'>>>;
+    user: Record<string, string>;
+  },
+  settings: {
+    user?: Record<string, UserProvidedValues<unknown>> | undefined;
+    defaults: Readonly<Record<string, Omit<UiSettingsParams, 'schema'>>>;
+  }
+): boolean {
+  const userTheme: string = userSettings?.user?.darkMode;
```

Comment:
Isn't retrieving the setting from a `UiSettingsUserClient` the same level of abstraction as the original code retrieving the value from the `UiSettingsClient`. They are sibling clients.

Reply:
It is, but it's not great. Ideally, the rendering service wouldn't communicate with the `UiSettingsClient` directly. The logic from lines 259 - 267 should be extracted to a service.

Reply:
That is a good idea, it would really simplify the code, but I think refactoring how the UISettingsClients is out of scope (I think that would probably involve more than the Rendering Service to be done properly?). Maybe it could be a FER or we could discuss it for the next iteration?